completions/jq

```diff
@@ -34,14 +34,32 @@ _jq()
                 ;;
         esac
 
+    # See whether the output of --help mentions --help
+    SEEMS_TO_HAVE_COMPLETE_HELP=$(jq --help | grep -q -e --help && echo yes || echo no)
```

Comment:
Let's also refactor this so that we grab the whole `--help` output here to a variable, and below when parsing help off it, just `printf` it and `_parse_help` from stdin instead of having to run `jq --help` again.

No need for the `grep` here then or a separate boolean var, can just `[[ $help_output == *--help* ]]` below.